### PR TITLE
Few more properly parsed files

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

Another part of #3611 with few more `TreeToIr` improvements.

### Important Notes

Unofficial `LoadParser.sh` check from #3611 of all library files now reports just 54 failures out of 222 files - e.g. 75% success rate.

### Checklist

Please include the following checklist in your PR:

- [x] All code conforms to the [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md)
- All code has been tested:
  - [x] Unit tests have been written where possible.

### CODE REVIEW

File: engine/runtime/src/test/java/org/enso/compiler/EnsoCompilerTest.java [hunk 1]
```diff
@@ -116,6 +130,23 @@ public void testFactorial() throws Exception {
...
+  @Test
+  public void testIsDigitWithSpaces() throws Exception {
+    parseTest("""
+    compare =
+        is_digit = character -> 42
+    """);
+  }
+
+  @Test
+  @Ignore
```
Comment (JaroslavTulach) [inline engine/runtime/src/test/java/org/enso/compiler/EnsoCompilerTest.java:142]: @kazcw, this test require changes on the Rust side. Presumably the `Tree` shall be identical to `testIsDigitWithSpaces`.

File: engine/runtime/src/test/java/org/enso/compiler/EnsoCompilerTest.java [hunk 2]
```diff
@@ -582,6 +624,15 @@ public void testExtensionMethod() throws Exception {
       """);
   }
 
+  @Test
+  @Ignore
+  public void testExtensionOperator() throws Exception {
+    parseTest("""
+    Text.* : Integer -> Text
```
Comment (JaroslavTulach) [inline engine/runtime/src/test/java/org/enso/compiler/EnsoCompilerTest.java:631]: Right now the combination of `.*` is represented as erroneous in the `Tree`. That will likely need some fixes on the Rust side, @kazcw.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Ordering/Natural_Order.enso [hunk 3]
```diff
@@ -29,7 +29,7 @@ compare text1 text2 case_sensitivity=Case_Sensitivity.Sensitive =
     ## check if a single character is between '0' and '9'
     ascii_code_zero = 48
     ascii_code_nine = 57
-    is_digit=character -> character>=ascii_code_zero && character<=ascii_code_nine
+    is_digit = character -> character>=ascii_code_zero && character<=ascii_code_nine
```
Comment (JaroslavTulach) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Ordering/Natural_Order.enso:32]: This change is covered by [following test](https://github.com/enso-org/enso/pull/3826#discussion_r1002676540), but easier to fix the syntax than wait for the test to be fixed.

**Review by jdunkerley — APPROVED** [review]
Approving the mini library change.

**Review by hubertp — APPROVED** [review]

**Pull request merged**